src/TwigComponent/src/CVA.php

```diff
@@ -12,107 +12,134 @@
-     * @var array<string, array<string, string|list<string|null>>|null the array should have the following format [variantCategory => [variantName => classes]]
-     *                                                ex: ['colors' => ['primary' => 'bleu-8000', 'danger' => 'red-800 text-bold'], 'size' => [...]]
-     * @var array<array<string, string|array<string>>> the array should have the following format ['variantsCategory' => ['variantName', 'variantName'], 'class' => 'text-red-500']
-     * @var array<string, string>|null
+     * @var list<string>
+     */
+    private readonly array $base;
+
+    /**
+     * @param string|list<string> $base The base classes to apply to the component
      */
     public function __construct(
-        private string|array|null $base = null,
-        private ?array $variants = null,
-        private ?array $compoundVariants = null,
-        private ?array $defaultVariants = null,
+        string|array $base = [],
+        /**
+         * The variants to apply based on recipes.
+         *
+         * Format: [variantCategory => [variantName => classes]]
+         *
+         * Example:
+         *      'colors' => [
+         *          'primary' => 'bleu-8000',
+         *          'danger' => 'red-800 text-bold',
+         *       ],
+         *      'size' => [...],
+         *
+         * @var array<string, array<string, string|list<string>>>
+         */
+        private readonly array $variants = [],
+
+        /**
+         * The compound variants to apply based on recipes.
+         *
+         * Format: [variantsCategory => ['variantName', 'variantName'], class: classes]
+         *
+         * Example:
+         *   [
+         *      'colors' => ['primary'],
+         *      'size' => ['small'],
+         *      'class' => 'text-red-500',
+         *   ],
+         *   [
+         *      'size' => ['large'],
+         *      'class' => 'font-weight-500',
+         *   ]
+         *
+         * @var array<array<string, string|array<string>>>
+         */
+        private readonly array $compoundVariants = [],
+
+        /**
+         * The default variants to apply if specific recipes aren't provided.
+         *
+         * Format: [variantCategory => variantName]
+         *
+         * Example:
+         *     'colors' => 'primary',
+         *
+         * @var array<string, string>
+         */
+        private readonly array $defaultVariants = [],
     ) {
+        $this->base = (array) $base;
     }
 
-    public function apply(array $recipes, ?string ...$classes): string
+    /**
+     * @deprecated since Symfony 2.17. Use {@see apply()} instead.
+     */
+    public function resolve(array $recipes): string
     {
-        return trim($this->resolve($recipes).' '.implode(' ', array_filter($classes)));
+        trigger_deprecation('symfony/ux-twig-component', '2.17', 'The method "%s()" is deprecated and will be remove in 3.0. Use "apply()" instead.', __METHOD__);
```

Comment:
I think we should just kill this method (and not worry about deprecating). I doubt anyone is using it.

Reply:
Good & thanks!